### PR TITLE
[Wizard] Provide a way to to cancel the provision of URL for additional sources

### LINKED ISSUE #1083: [Wizard] Be able to cancel the provision of URL for additional sources
In the wizard, in the second step:
```julia
Would you like to download additional sources?  [y/N]: y
Please enter a URL (git repository or compressed archive) containing the source code to build:
```

if one selects `y`, then there is no way to backtrack, even writing a wrong URL will re-ask for a good one:

```julia
julia> BinaryBuilder.run_wizard()
Would you like to resume the previous incomplete wizard run?
 > Resume previous run
   Start from scratch
			# Step 2a: Obtain the source code

Please enter a URL (git repository or compressed archive) containing the source code to build:
> 

URL may not be empty!
Please enter a URL (git repository or compressed archive) containing the source code to build:
> 

URL may not be empty!
Please enter a URL (git repository or compressed archive) containing the source code to build:
> 

URL may not be empty!
Please enter a URL (git repository or compressed archive) containing the source code to build:
> 

URL may not be empty!
Please enter a URL (git repository or compressed archive) containing the source code to build:
> ^C

Wizard stopped, use BinaryBuilder.run_wizard() to resume

WizardState [step2]
```

### PR DESCRIPTION
**Pull request by gbaraldi**
Fixes #1083. I'm not satisfied with the wording but it does work. Probably could do with some tests but I'm not sure how to test things in the wizard

### CODE REVIEW

File: src/wizard/obtain_source.jl [hunk 1]
```diff
@@ -115,7 +117,7 @@ function download_source(state::WizardState)
             continue
         end
     end
-
+
```
Comment (giordano) [inline src/wizard/obtain_source.jl:120]:
```suggestion

```

**Review by giordano — COMMENTED** [review]
> Probably could do with some tests but I'm not sure how to test things in the wizard

Well, welcome on board :grin:  You can have a look at [`test/wizard.jl`](https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/master/test/wizard.jl), you have to simulate all inputs :slightly_smiling_face:

**Comment by giordano** [comment]
@gbaraldi are you still planning to add tests? :slightly_smiling_face:

**Comment by gbaraldi** [comment]
I completely forgot about this PR I will add tests this week then. 😅

**Comment by giordano** [comment]
Lovely, thanks!

**Pull request merged**